### PR TITLE
Wait after restarting FTL before trying to check version

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Prevent errors like seen in https://github.com/pi-hole/pi-hole/issues/5608#issuecomment-2017000214 when Pi-hole uses itself as `nameserver`:

```bash
 Please note that changing branches severely alters your Pi-hole subsystems
  Features that work on the master branch, may not on a development branch
  This feature is NOT supported unless a Pi-hole developer explicitly asks!
  Have you read and understood this? [y/N] y

  [✓] Branch fix/ede_neterr exists
  [i] Switching to branch: "fix/ede_neterr" from "fix/ede_neterr"
  [✓] Downloading and Installing FTL
  [✓] Restarting pihole-FTL service...
  [✓] Enabling pihole-FTL service to start on reboot...
fatal: unable to access 'https://github.com/pi-hole/pi-hole/': Could not resolve host: github.com
fatal: unable to access 'https://github.com/pi-hole/web/': Could not resolve host: github.com
fatal: unable to access 'https://github.com/pi-hole/FTL/': Could not resolve host: github.com

```

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: advanced/Scripts/piholeCheckout.sh [hunk 1]
```diff
@@ -170,10 +170,23 @@ checkout() {
...
+            echo -e "  ${INFO} Switching to branch \"${2}\" from \"${oldbranch}\""
             FTLinstall "${binary}"
             restart_service pihole-FTL
             enable_service pihole-FTL
+            str="Restarting FTL..."
+            echo -ne "  ${INFO} ${str}"
+            # Wait until name resolution is working again after restarting FTL,
+            # so that the updatechecker can run successfully and does not fail
+            # trying to resolve github.com
+            until getent hosts github.com &> /dev/null; do
```
Comment (rdwebdesign) [inline advanced/Scripts/piholeCheckout.sh:182]: Is it possible `enable_service pihole-FTL` fails (or resolution fails) and we ended up in an infinite loop?

Comment (PromoFaux) [reply advanced/Scripts/piholeCheckout.sh:182]: I think it makes sense to loop until github.com is resolvable - if the network is down then there is no harm in waiting for it to be up again.

**Review by yubiuser — COMMENTED** [review]
I agree with rdwebdesign. We should be not have an infinite loop that depends on outside conditions (network availability). I think we should have a counter and exit the loop after e.g. 10 seconds and ask users to manually run the update command afterwards.

**Comment by PromoFaux** [comment]
I am not sure where my mind was when I first responded to RD's comment. Reading over the code again, actually it probably is a fair shout that we timeout after x seconds (the user could always `ctrl+c`, but that is not overly friendly

**Comment by DL6ER** [comment]
The script is constantly adding `.` to the output to show it's still alive. Waiting "forever" seems fine to me as the updater has no way to finish properly when there in no way to contact GitHub.

The suggested alternative is erroring out with no option for the user to wait longer if 10 seconds isn't enough to restart FTL on their RPi Zero with 100,000 queries to import from the database. Sure, the timeout could be 60 seconds but my opinion is that waiting "forever" and constantly showing that the script is still alive is the proper way to do this

**Comment by yubiuser** [comment]
I know we had this before, but maybe the answer changed in-between: do we have any signal that would signal FTL's readiness to accept DNS queries (e.g. a second pid file)? 
___
This PR is not about FTL only. The cat might have unplugged the cable just at the right moment - FTL might be ready for a long time, but there would be no network connectivity.

**Comment by DL6ER** [comment]
We could `dig pi.hole` or whatever always resolves without being forwarded.

I hear you and you are right, it's about more than FTL alone. However, without connectivity, there is no (simple) way the updater can exit gracefully.

**Comment by yubiuser** [comment]
`dig pi.hole @localhost` could work. But this tells us only that FTL is ready, but has no meaning for systems that don't use FTL for their host's DNS resolution.

**Comment by DL6ER** [comment]
Quite some time has passed and I'd like to re-inquire how we all feel about this PR. After reading up on its motivation, it seems still very relevant on slow devices. The test currently tests for resolvability of GitHub which makes sense because the next step is querying the versions from GitHub.

Hence, I also feel like the question we should be asking is **not** "is FTL ready" but **rather** "can the host resolve github.com" - which is exactly what the code in the PR wants to implement.

I did just rebase this on latest `development-v6`.

**Comment by rdwebdesign** [comment]
CentOS 9 tests are failing.

**Comment by DL6ER** [comment]
Rebased based on https://github.com/pi-hole/pi-hole/pull/5743 to check effect on CI tests

**Comment by DL6ER** [comment]
GitHub forced me to update the branch *again* after merging the other PR. Nothing changed - except, of course, `HEAD^` hash so we get a new commit hash here, too.

**Review by yubiuser — APPROVED** [review]
We also have a similar check if updating `core`: this will trigger the installer, which will run `gravity` right before `updatechecker.sh`. `Gravity` contains a check which tries to resolve `raw.githubusercontent.com` 4 times before failing.

See https://github.com/pi-hole/pi-hole/pull/5137 and
https://github.com/pi-hole/pi-hole/blob/a965afa249740e8f4803d999571b1df69d5eaa21/gravity.sh#L296

**Pull request merged**